lib/charms/layer/basic.py

```diff
@@ -169,11 +169,43 @@ def apt_install(packages):
             break
 
 
+def _patch_config_cached():
+    """
+    For performance reasons, the update-status hook was intended to be as
+    lightweight as possible and ideally to not make any API calls to the
```

Comment:
There does need to be one api call - the status-set hook tool to actually update the status